packages/read-model/src/Product/Detail/ProductDetailViewElasticsearchFactory.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\ReadModelBundle\Product\Detail;
+
+use Shopsys\FrameworkBundle\Component\Utils\Utils;
+use Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\FrameworkBundle\Model\Seo\SeoSettingFacade;
+use Shopsys\ReadModelBundle\Brand\BrandView;
+use Shopsys\ReadModelBundle\Brand\BrandViewFactory;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Image\ImageViewFacadeInterface;
+use Shopsys\ReadModelBundle\Parameter\ParameterViewFactory;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionViewFactory;
+use Shopsys\ReadModelBundle\Product\PriceFactory;
+
+class ProductDetailViewElasticsearchFactory
```

Comment:
I don't know. I think only one factory should exist... but maybe if we deprecate database behavior and remove the whole DB layer from the read model, everything will be fine in the end.

Reply:
I am really convinced there should be separate factories for creating the same object from different data.

Maybe it is confusing the `factory` in class name, because this pattern is more like an adapter.

Reply:
okay. I accept that